receiver/prometheusreceiver/internal/prom_to_otlp_test.go

```diff
@@ -57,76 +59,151 @@ func makeK8sResource(jobInstance *jobInstanceDefinition, def *k8sResourceDefinit
 }
 
 func makeResourceWithJobInstanceScheme(def *jobInstanceDefinition, hasHost bool) pcommon.Resource {
+	resource := pcommon.NewResource()
+	attrs := resource.Attributes()
+	// Using hardcoded values to assert on outward expectations so that
+	// when variables change, these tests will fail and we'll have reports.
+	attrs.PutStr("service.name", def.job)
+	if hasHost {
+		attrs.PutStr("server.address", def.host)
+	}
+	attrs.PutStr("service.instance.id", def.instance)
+	attrs.PutStr("server.port", def.port)
+	attrs.PutStr("url.scheme", def.scheme)
```

Comment:
What happens if the scraped metrics already have these attributes, has this been discussed when developing the spec? Shouldn't the resource attributes for a scrape endpoint be in the `telemetry.*` prefix instead? Sorry, this is the wrong place to discuss spec, but maybe the right people are here for s start

Reply:
Today, it isn't possible for prometheus metrics to have attributes which have dots, so collisions are not currently possible.

Reply:
The `target_info` prometheus metric is converted to the OpenTelmetry resource, so there will be the possibility of collision in the future.  In that case, I think we should use `honor_labels` in the prom config to determine which takes precedence.

Reply:
I do wonder if these "general attributes" should actually be added as resource attributes at all...

Reply:
Resolving, as I don't think this blocks this PR.